src/infi/clickhouse_orm/fields.py

```diff
@@ -70,8 +70,9 @@ def get_sql(self, with_default=True):
         else:
             return self.db_type
 
-    def is_insertable(self):
-        return self.alias is None and self.materialized is None
+    @property
+    def readonly(self):
+        return self.alias is not None or self.materialized is not None
```

Comment:
To my mind, materialized ="" and materialized=None is not the same thing.
All of them will lead to readonly = True in your example. May be, it should be filtered by validation in __init__, but from my point of view it's more readable, then without type specifying (I catched bugs in other python projects, with [], "" considered as False...)